src/networkpolicy/networkPolicy.go

```diff
@@ -1858,6 +1893,49 @@ func convertKnoxNetworkLogToKnoxNetworkPolicy(log *types.KnoxNetworkLog, pods []
 			ePolicy.Metadata["namespace"] = log.SrcNamespace
 			egressPolicy = &ePolicy
 		}
+	} else if log.DstPodName == "" && len(log.DstReservedLabels) == 0 && log.Direction == "EGRESS" {
+		// Egress Policy
+		ePolicy := buildNewKnoxEgressPolicy()
+		egress := types.Egress{}
+
+		ePolicy.Spec.Selector.MatchLabels = getEndpointMatchLabels(log.SrcPodName, pods)
+
+		var cidrs []string
+		cidrs = append(cidrs, "0.0.0.0/32")
+
+		egress.ToCIDRs = append(egress.ToCIDRs, types.SpecCIDR{
+			CIDRs: cidrs,
+		})
+
+		egress.ToPorts = append(egress.ToPorts, types.SpecPort{
+			Port:     strconv.Itoa(log.DstPort),
+			Protocol: libs.GetProtocol(log.Protocol),
+		})
+
+		ePolicy.Spec.Egress = append(ePolicy.Spec.Egress, egress)
+		ePolicy.Metadata["namespace"] = log.SrcNamespace
+		egressPolicy = &ePolicy
+	} else if log.DstPodName == "" && len(log.DstReservedLabels) == 0 && log.Direction == "INGRESS" {
```

Comment:
There is too much of duplicate code in the two condition. Can we create a function to handle this?